projects/ngx-sub-form/src/lib/ngx-automatic-root-form.component.ts

```diff
@@ -0,0 +1,40 @@
+import { NgxRootFormComponent } from './ngx-root-form.component';
+import { Observable, timer } from 'rxjs';
+import { debounce } from 'rxjs/operators';
+import { isEqual } from 'lodash-es';
+
+export const NGX_SUB_FORM_HANDLE_DATA_OUTPUT_STRATEGIES = {
+  debounce: (time: number): ReturnType<NgxAutomaticRootFormComponent<any, any>['handleDataOutput']> => obs =>
+    obs.pipe(debounce(() => timer(time))),
+};
+
+export abstract class NgxAutomaticRootFormComponent<
+  ControlInterface,
+  FormInterface = ControlInterface
+> extends NgxRootFormComponent<ControlInterface, FormInterface> {
+  public ngOnInit(): void {
+    super.ngOnInit();
+
+    this.registerOnChange(data => {
+      if (this.formGroup.invalid || isEqual(data, this.dataInput)) {
+        return;
+      }
+
+      this.dataValue = data;
+
+      this.applyingChangeSignal.next(true);
+
+      if (this.formGroup) {
+        this.formGroup.markAsPristine();
+
+        if (this.formGroup.valid) {
+          this._dataOutput$.next(data);
+        }
+      }
+    });
+  }
+
+  protected abstract handleDataOutput(): (
```

Comment:
Given the default behavior (immediate emit) is acceptable I think we shouldn't enforce implementation of this

Reply:
Agree. Fixed.